src/components/MultiSelect/helpers/normalizeValue.js

```diff
@@ -0,0 +1,18 @@
+export default function normalizeValue(value) {
+    if (!value || typeof value !== 'object') {
+        return [];
+    }
+    if (Array.isArray(value)) {
+        if (value.length === 0) {
+            return [];
+        }
+        return value.map(v => {
+            const newValue = v;
+            delete newValue.icon;
+            return newValue;
+        });
```

Comment:
let's stick to functional programming, here you are coping the v reference to newValue and then when doing ```delete newValue.icon;``` there is a mutation of the newValue and thus to the original v object, then instead let's do:
```
return value.map((item) => {
    const { value, label, name  } = item;  // here we control exactly what we want to pass
    return {
        value,
        label,
        name,
    };
});
```